rubi/rubi/data/market.py

```diff
@@ -1,394 +1,520 @@
-from typing import Optional, Dict, Union
+import logging as log
+from _decimal import Decimal
+from datetime import datetime
+from typing import Optional, Dict, List, Union
 
 import pandas as pd
 from eth_typing import ChecksumAddress
-from subgrounds import Subgrounds
+from subgrounds import Subgrounds, Subgraph, SyntheticField
+from subgrounds.pagination import ShallowStrategy
+from web3 import Web3
 
-from rubi.contracts import (
-    ERC20,
-)
-from rubi.network import (
-    Network,
-)
-from rubi.rubicon_types import (
-    OrderSide,
-    OrderQuery,
-    TradeQuery,
-)
+from rubi.contracts import ERC20
+from rubi.data.helpers import QueryValidation
+from rubi.data.helpers import SubgraphOffer, SubgraphTrade
+
+logger = log.getLogger(__name__)
+
+# Stop subgrounds from logging kak
+log.getLogger("subgrounds").setLevel(log.WARNING)
 
 
 class MarketData:
     """This class represents the RubiconV2 Subgraph, which contains data primarily related to the RubiconMarket.sol
-    contract. If a Network object is not passed in instantiation then this class will only be used to query data related
-     to the subgraph.
-
-    :param subgrounds: a Subgrounds instance
-    :type subgrounds: Subgrounds
-        param subgraph_url: a RubiconV2 Subgraph url endpoint that should be utilized for this class
-    :type subgraph_url: str
-    :param network: a Network object, native to the package
-    :type network: Network
+    contract.
+
+    :param url: a RubiconV2 Subgraph url endpoint that should be utilized for this class
+    :type url: str
+    :param fallback_url: a fallback url if we fail to connect to the primary url
+    :type fallback_url: str
+    :param tokens: A dictionary of token addresses and symbols to their ERC20 instantiation
+    :type tokens: Dict[ChecksumAddress | str, ERC20]
     """
 
     def __init__(
         self,
-        subgrounds: Subgrounds,
-        subgraph_url: str,
-        subgraph_fallback_url: str,
-        network: Optional[Network] = None,
-        network_tokens: Optional[Dict[ChecksumAddress, ERC20]] = None,
+        url: str,
+        fallback_url: str,
+        tokens: Dict[Union[ChecksumAddress, str], ERC20],
```

Comment:
i believe we will want this to be an optional parameter, that way users can access subgraph data without the need to have connection to a node (worth noting by doing so they will lose the ability to have formatted data)